x/fswap/keeper/genesis.go

```diff
@@ -0,0 +1,31 @@
+package keeper
+
+import (
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/x/fswap/types"
+)
+
+// InitGenesis initializes the module's state from a provided genesis
+// state.
+func (k Keeper) InitGenesis(ctx sdk.Context, bk types.BankKeeper, genState types.GenesisState) {
+	if err := k.SetParams(ctx, genState.Params); err != nil {
+		panic(err)
+	}
+	if err := k.SetSwapped(ctx, genState.Swapped); err != nil {
+		panic(err)
+	}
+	totalOldCoinsSupply := bk.GetSupply(ctx, types.DefaultOldCoins).Amount
+	totalNewCoinsSupply := types.DefaultSwapRate.MulInt(totalOldCoinsSupply)
+	totalNewCoins := sdk.NewDecCoinFromDec(genState.Params.NewCoinDenom, totalNewCoinsSupply)
+	if err := k.SetTotalSupply(ctx, totalNewCoins); err != nil {
```

Comment:
Is the total supply a read only value? If so, it would be fine, or you may add a comment on the relevant line (on keys.go?).
If not, you should consider exporting it on `ExportGenesis()`.

Reply:
`Total supply` is a read-only value and does not change once the swap is activated. I will add a comment in this section of the codes and include a line in keys.go to mention this.

Reply:
Is it OK to change the value of the total supply after exporting and importing the genesis? We should always consider the scenario.